packages/terra-application-layout/docs/usage.md

```diff
@@ -0,0 +1,258 @@
+
+The `extensions` prop allows consumers to render an element within the ApplicationLayout's extensions region. The `extensions` content will be rendered in various ways depending on the current breakpoint. The `extensions` element will receive `app` and `layoutConfig` props from the ApplicationLayout, allowing it to interact with and respond to changes within the ApplicationLayout.
+
+```jsx
+import { Utils } from 'terra-application-layout';
+
+const MyExtensions = ({ app, layoutConfig }) => {
+  if (Utils.helpers.isSizeCompact(layoutConfig.size)) {
+    return (
+      <div>Extensions for ApplicationLayout menu!</div>
+    );
+  } else {
+    return (
+      <div>Extensions for ApplicationLayout header!</div>
+    );
+  }
+};
+```
+
+### `indexPath`
+#### Is Required: `true`
+
+The `indexPath` prop allows consumers to set the default path of the ApplicationLayout. The ApplicationLayout will redirect to this path should users of the consuming application attempt to route to a component not detailed in the `routingConfig` prop. Accordingly, the `indexPath` value must have an associated entry within the `routingConfig` specification.
+
+
+### `nameConfig`
+#### Is Required: `false`
+
+The `nameConfig` prop allows consuming applications to add their own branding to the ApplicationLayout. The ApplicationLayout will render this content in different ways based on the active responsive breakpoint.
+
+|Key Name|Type|Is Required|Description|
+|---|---|---|---|
+|`accessory`|Element|optional|A React element that will be presented beside the title. This is typically some sort of icon or visual branding.
+|`title`|String|optional|The title of the application. This text will be styled by the ApplicationLayout.|
+
+```jsx
+const myNameConfig = {
+  accessory: <MyApplicationLogo />,
+  title: 'My Application',
+}
+```
+
+### `navigationItems`
+#### Is Required: `false`
+
+The `navigationItems` prop allows consumers to render high-level, primary navigation controls directly within the ApplicationLayout. The ApplicationLayout will render this content in different ways based on the active responsive breakpoint. If `navigationItems` are omitted, no primary navigation controls will be rendered.
+
+Each navigation item provided must be associated to a path that is present within the `routingConfig` specification. The navigation item, as rendered by the ApplicationLayout, will route to that path upon selection.
+
+The value provided for `navigationItems` should be an array of objects with the following API:
+
+|Key Name|Type|Is Required|Description|
+|---|---|---|---|
+|`text`|String|**required**|A string rendered within the navigation item control.|
+|`path`|Element|**required**|A string path to route to upon navigation item selection. This path should be present within the `routingConfig` as well.|
+
+```jsx
+const navigationItems = [{
+  path: '/page_1',
+  text: 'Page 1',
+}, {
+  path: '/page_2',
+  text: 'Page 2',
+}];
+```
+
+### `routingConfig`
```

Comment:
Talked with @tbiethman offline. I'd like to see the routingConfig doc detailed some more on the shape of the object it takes.  e.g. Add more descriptions for what is `content`, what is `menu`, what is the `path`, `component`, `default`, and `component class` keys. This is available if you navigate to the terra-navigation-layout docs, but I think it would be helpful to consumers to describe this object here as well.